src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -135,25 +134,37 @@ public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
             while (sourceIndex < srcLength - 2)
             {
                 result = Encode(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex + 4 > destLength) goto False;
                 Unsafe.WriteUnaligned(ref Unsafe.Add(ref destBytes, destIndex), result);
-                sourceIndex += 3;
                 destIndex += 4;
+                sourceIndex += 3;
             }
 
             if (sourceIndex == srcLength - 1)
             {
                 result = EncodePadByTwo(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex + 4 > destLength) goto False;
                 Unsafe.WriteUnaligned(ref Unsafe.Add(ref destBytes, destIndex), result);
                 destIndex += 4;
+                sourceIndex += 1;
             }
             else if (sourceIndex == srcLength - 2)
             {
                 result = EncodePadByOne(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex + 4 > destLength) goto False;
                 Unsafe.WriteUnaligned(ref Unsafe.Add(ref destBytes, destIndex), result);
                 destIndex += 4;
+                sourceIndex += 2;
             }
 
-            return destIndex;
+            bytesConsumed = sourceIndex;
+            bytesWritten = destIndex;
+            return true;
+
+            False:
```

Comment:
How about FalseExit? I don't know if ErrorExit is appropriate since it isn't necessarily an error condition.